source/type/pb_converter.h

```diff
@@ -58,17 +72,329 @@ namespace navitia {
+inline pbnavitia::NavitiaType get_embedded_type(const ng::Admin*) { return pbnavitia::ADMINISTRATIVE_REGION; }
+inline pbnavitia::NavitiaType get_embedded_type(const nt::StopArea*) { return pbnavitia::STOP_AREA; }
+inline pbnavitia::NavitiaType get_embedded_type(const nt::StopPoint*) { return pbnavitia::STOP_POINT; }
+inline pbnavitia::NavitiaType get_embedded_type(const nt::CommercialMode*) { return pbnavitia::COMMERCIAL_MODE; }
+inline pbnavitia::NavitiaType get_embedded_type(const nt::MetaVehicleJourney*) { return pbnavitia::TRIP; }
+
+inline pbnavitia::Calendar* get_sub_object(const nt::Calendar*,
+                                           pbnavitia::PtObject* pt_object) { return pt_object->mutable_calendar(); }
+inline pbnavitia::VehicleJourney* get_sub_object(const nt::VehicleJourney*,
+                                                 pbnavitia::PtObject* pt_object) { return pt_object->mutable_vehicle_journey(); }
+inline pbnavitia::Line* get_sub_object(const nt::Line*,
+                                       pbnavitia::PtObject* pt_object) { return pt_object->mutable_line(); }
+inline pbnavitia::Route* get_sub_object(const nt::Route*,
+                                        pbnavitia::PtObject* pt_object) { return pt_object->mutable_route(); }
+inline pbnavitia::Company* get_sub_object(const nt::Company*,
+                                          pbnavitia::PtObject* pt_object) { return pt_object->mutable_company(); }
+inline pbnavitia::Network* get_sub_object(const nt::Network*,
+                                          pbnavitia::PtObject* pt_object) { return pt_object->mutable_network(); }
+inline pbnavitia::Poi* get_sub_object(const ng::POI*,
+                                      pbnavitia::PtObject* pt_object) { return pt_object->mutable_poi(); }
+inline pbnavitia::AdministrativeRegion* get_sub_object(const ng::Admin*,
+                                                       pbnavitia::PtObject* pt_object) { return pt_object->mutable_administrative_region(); }
+inline pbnavitia::StopArea* get_sub_object(const nt::StopArea*,
+                                           pbnavitia::PtObject* pt_object) { return pt_object->mutable_stop_area(); }
+inline pbnavitia::StopPoint* get_sub_object(const nt::StopPoint*,
+                                            pbnavitia::PtObject* pt_object) { return pt_object->mutable_stop_point(); }
+inline pbnavitia::CommercialMode* get_sub_object(const nt::CommercialMode*,
+                                                 pbnavitia::PtObject* pt_object) { return pt_object->mutable_commercial_mode(); }
+inline pbnavitia::Trip* get_sub_object(const nt::MetaVehicleJourney*,
+                                       pbnavitia::PtObject* pt_object) { return pt_object->mutable_trip(); }
+
+namespace {
+template<typename T> nt::Type_e get_type_e() {
+    static_assert(!std::is_same<T, T>::value, "get_type_e unimplemented");
```

Comment:
`!std::is_same<T, T>::value` always returns false, right? why not using false instead?